consensus/bor/span_store.go

```diff
@@ -46,34 +56,181 @@ func (s *SpanStore) spanById(ctx context.Context, spanId uint64) (*span.Heimdall
 		currentSpan, _ = value.(*span.HeimdallSpan)
 	}
 
-	if currentSpan == nil {
-		var err error
-		if s.heimdallClient == nil {
-			if spanId == 0 {
-				currentSpan, err = getMockSpan0(ctx, s.spanner, s.chainId)
-				if err != nil {
-					log.Warn("Unable to fetch span from heimdall", "id", spanId, "err", err)
-					return nil, err
-				}
-			} else {
-				return nil, fmt.Errorf("unable to create test span without heimdall client for id %d", spanId)
-			}
-		} else {
-			currentSpan, err = s.heimdallClient.Span(ctx, spanId)
+	if currentSpan != nil {
```

Comment:
The `spanById` looks a very complicated function. Not sure if it's worth refactoring, but I understand the hassle for testing and all...

Reply:
This logic will be deleted with the veblop release.